File: source/android/uitestapp/src/main/java/io/adaptivecards/uitestapp/ui/rendered_card/RenderedCardViewModel.java
package io.adaptivecards.uitestapp.ui.rendered_card;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import java.util.List;

import io.adaptivecards.uitestapp.ui.inputs.RetrievedInput;

public class RenderedCardViewModel extends ViewModel
{
    private MutableLiveData<List<RetrievedInput>> mInputs;

    public RenderedCardViewModel()
    {
        mInputs = new MutableLiveData<>();
    }

    public MutableLiveData<List<RetrievedInput>> getInputs()
    {
        return mInputs;
    }
}

File: source/android/uitestapp/src/main/java/io/adaptivecards/uitestapp/ui/rendered_card/RenderedCardViewModel.kt
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
package io.adaptivecards.uitestapp.ui.rendered_card

import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel
import io.adaptivecards.uitestapp.ui.inputs.RetrievedInput

import androidx.test.espresso.idling.CountingIdlingResource

class RenderedCardViewModel : ViewModel() {
    val inputs: MutableLiveData<List<RetrievedInput>>
    val idlingResource: MutableLiveData<CountingIdlingResource>

    init {
        inputs = MutableLiveData()
        idlingResource = MutableLiveData(CountingIdlingResource("filteredChoiceSetIdlingResource"))
    }
}
